### PR TITLE
OpenACC schedule type in parallel_for and parallel_reduce (RangePolicy)

### PR DESCRIPTION
**Pull request by dalg24**
Implement Schedule<Static> in parallel_for and parallel_reduce and use dynamic scheduling by default

### CODE REVIEW

**Review by seyonglee — APPROVED** [review]
It looks good to me.
The only possible issue is that it is unknown whether Kokkos dynamic schedule will be observed by the OpenACC compiler or not. OpenACC does not provide any method to enforce the OpenACC implementation to use OpenMP-style dynamic scheduling (and with specific chunk size).

**Comment by seyonglee** [comment]
There are two options that we can do for the Kokkos dynamic schedule: 
Option 1) just let the OpenACC implementation do whatever it can do - no guarantee on dynamic scheduling and chunk size.
Option 2) use the same implementation as the Kokkos static schedule, which will guarantee the chunk size but loop will be statically scheduled.

**Review by JBludau — APPROVED** [review]
looks good given the GPU converts the `(static: chunk_size)` into dynamic scheduling when we have more gangs than SMs ...

**Review by pedrovalerolara — APPROVED** [review]

**Comment by dalg24** [comment]
Retest this please

**Comment by seyonglee** [comment]
I see several compilation errors when testing the latest version; do not merge this PR until those errors are fixed.

**Comment by seyonglee** [comment]
> I see several compilation errors when testing the latest version; do not merge this PR until those errors are fixed.

The compilation errors are fixed.

**Review by seyonglee — APPROVED** [review]

File: core/src/OpenACC/Kokkos_OpenACC_ParallelFor_Range.hpp [hunk 1]
```diff
@@ -54,7 +54,10 @@ namespace Kokkos::Experimental::Impl {
 template <class IndexType, class Functor>
 void OpenACCParallelForRangePolicy(Schedule<Static>, int chunk_size,
                                    IndexType begin, IndexType end,
-                                   Functor functor, int async_arg) {
+                                   Functor afunctor, int async_arg) {
+  // Create a local copy of the functor to avoid errors due to incorect scope
+  // analysis.
```
[the remark this answers is not in the document]
Comment (seyonglee) [reply core/src/OpenACC/Kokkos_OpenACC_ParallelFor_Range.hpp:59]: It is a kind of compiler bug; the NVHPC compiler incorrectly assumes that the functor object is shared, and thus it is passed to the device as if a pointer type whose size is unknown. NVHPC 22.7 still has this problem.

[the remark this answers is not in the document]
Comment (dalg24) [reply core/src/OpenACC/Kokkos_OpenACC_ParallelFor_Range.hpp:59]: Then I'd say let us capture the compile error and add greppable markers, something like
```
// FIXME_OPENACC FIXME_NVHPC workaround compiler bug (incorrect scope analysis)
// NVC++-S-1067-Cannot determine bounds for array - functor

[the remark this answers is not in the document]
Comment (seyonglee) [reply core/src/OpenACC/Kokkos_OpenACC_ParallelFor_Range.hpp]: Updated the comments.

**Comment by dalg24** [comment]
One HIP build headed straight to timeout.  Safe to ignore.

**Pull request merged**